### ISSUE TITLE
Memory increased when using a larger training dataset

### ISSUE THREAD

**Issue by TrinhQuocNguyen** [TAMU-VITA/EnlightenGAN#45]
Hi TAMU-VITA,
Thank you for awesome work.
I have tried to train the model on a larger dataset, but the RAM memory was increasing during the time of loading data. Then as the consequence, it's got killed.

Is there any place in the dataloader class that we can modify to avoid above problem?

Thank you.

**Comment by TrinhQuocNguyen** [comment]
I have modified the source code and can able to load any huge dataset.
Thank you.

**Comment by zhangxiaopang88** [comment]
您好，请问您修改了哪部分源码？您构建大数据集时是怎么构建的？希望得到您的回复 @TrinhQuocNguyen

**Comment by JustinLiu97** [comment]
> I have modified the source code and can able to load any huge dataset.
> Thank you.

Hi Trinh,

I am observing the same issue when training and testing on original data. The testing on original test set can be completed with 300GB of swap memory, and the training will get killed after several epochs under this setting.

Is it possible to share your way of solving it?

Thanks!
